src/gt4py/backend/python_generator.py

```diff
@@ -261,6 +261,7 @@ def visit_Stage(self, node: gt_ir.Stage, *, iteration_order):
         self.block_info.accessors = {accessor.symbol for accessor in node.accessors}
         self.block_info.iteration_order = iteration_order
         self.block_info.extent = node.compute_extent
+        self.var_refs_defined.clear()
```

Comment:
Was this change intended?

Reply:
Yes, this change is necessary for conditionals to work properly in the `numpy` backend. Will add a test for that case.